lib/gooddata/helpers/data_helper.rb

```diff
@@ -115,12 +115,13 @@ def realize_link
       end
 
       def realize_s3(params)
-        s3_client = params['aws_client'] && params['aws_client']['s3_client']
+        s3_client = params['s3_client'] && params['s3_client']['client']
         raise 'AWS client not present. Perhaps S3Middleware is missing in the brick definition?' if !s3_client || !s3_client.respond_to?(:bucket)
         bucket_name = @options[:bucket]
```

Comment:
Do we still read bucket from input source?

Reply:
Yes, as in my previous answer, we have to use bucket in downloading file.